rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,538 @@
+> _Example: `14:30:00` for 2:30PM or `25:35:00` for 1:35AM on the next day._
+
+### Trip Key
+For scheduled trips, Glides does not have the same trip ID used by RTR. For added trips, while an ID is provided, it can be the case that an added trip from the Glides perspective is matched to a GTFS trip, in which case RTR will use the GTFS trip ID in publishing the information.
+
+In order to reduce message duplication, a Trip Key is used to identify both added and scheduled trips.
+
+**Added Trip**
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `id` (string): unique value representing this trip for the given `serviceDate`
+```json
+`{"serviceDate": "2023-01-20", "id": "ADDED-123"}
+```
+**Scheduled Trip (departure time from origin)
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `departureTime` (Time): when the vehicle is scheduled to depart the given location
+- `startLocation` (Location): where the trip is departing from
+```json
+`{"serviceDate": "2023-01-20", "departureTime": "25:01:02", "startLocation": "gtfs:place-lake"}
+```
+**Scheduled Trip (arrival time at destination)
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `arrivalTime` (Time): when the vehicle is scheduled to arrive at the given location
+- `endLocation` (Location): where the trip is ending
+```json
+`{"serviceDate": "2023-01-20", "arrivalTime": "09:10:11", "endLocation": "id:yard-inner"}
+```
+
+### Trip Type
+When adding a trip, there are three kinds of trips which can be added:
+- `"arrival"`: a trip arriving to `endLocation` at `time`
+- `"departure"`: a trip departing from `startLocation` at `time`
+
+## Events
+### Operator signed in
+At the start of their shift, operators need to confirm that they are fit-for-duty and do not have any electronic devices. They do this by either physically signing a paper trainsheet, or by typing their badge number into Glides. This event represent either type of signin, but may not be generated in the case where an operator only signs into the paper trainsheet.
+
+Event type: `com.mbta.ctd.glides.operator-signed-in`
+Fields in the event:
+- `author` (Author): the inspector who signed in the operator
+- `operator` (Operator): the operator who signed in
+- `signedInAt` (RFC3999 timestamp): the time at which they signed in (separate from the `time` of the event)
+
+### Batched
+If multiple events are generated as a part of the same author action, then they can be combined into a single Batched event. This ensures that all of the events are kept together throughout the data pipeline.
+
+Event type: `com.mbta.ctd.glides.batched`
+Fields in the event:
+- `author` (Author): the inspector who performed the events
+- `events` (array of `{"type": string, "data": JSON}`): each element in the array is a CloudEvent including only the `type` and `data` fields. See [Examples][#examples] for an example.
```

Comment:
Do the batched events need to re-include the author? It's not clear cuz the author is elided in the examples.